Source/Core/VideoCommon/PixelShaderGen.cpp

```diff
@@ -796,23 +898,19 @@ static void WriteStage(T& out, pixel_shader_uid_data* uid_data, int n, API_TYPE
 	   ac.a == TEVALPHAARG_RASA || ac.b == TEVALPHAARG_RASA ||
 	   ac.c == TEVALPHAARG_RASA || ac.d == TEVALPHAARG_RASA)
 	{
-		const int i = bpmem.combiners[n].alphaC.rswap;
-		uid_data->stagehash[n].ac |= bpmem.combiners[n].alphaC.rswap;
-		uid_data->stagehash[n].tevksel_swap1a = bpmem.tevksel[i*2].swap1;
-		uid_data->stagehash[n].tevksel_swap2a = bpmem.tevksel[i*2].swap2;
-		uid_data->stagehash[n].tevksel_swap1b = bpmem.tevksel[i*2+1].swap1;
-		uid_data->stagehash[n].tevksel_swap2b = bpmem.tevksel[i*2+1].swap2;
-		uid_data->stagehash[n].tevorders_colorchan = bpmem.tevorders[n / 2].getColorChan(n & 1);
-
-		const char *rasswap = swapModeTable[bpmem.combiners[n].alphaC.rswap];
-		out.Write("\trastemp = %s.%s;\n", tevRasTable[bpmem.tevorders[n / 2].getColorChan(n & 1)], rasswap);
+		char rasswap[5] = {
```

Comment:
I'm skeptical of this bit of code.
![image](https://cloud.githubusercontent.com/assets/1018829/12372882/3185241e-bc2c-11e5-915f-9f1d4e766cd9.png)